lib/common/results.c

```diff
@@ -876,3 +876,47 @@ pcmk__reset_result(pcmk__action_result_t *result)
     free(result->action_stderr);
     result->action_stderr = NULL;
 }
+
+/*!
+ * \internal
+ * \brief Copy the result of an action
+ *
+ * \param[in]  src  Result to copy
+ * \param[out] dst  Where to copy \p src to
+ */
+void
+pcmk__copy_result(pcmk__action_result_t *src, pcmk__action_result_t *dst)
+{
+    if ((src == NULL) || (dst == NULL)) {
+        return;
+    }
+    dst->exit_status = src->exit_status;
+    dst->execution_status = src->execution_status;
+
+    if (!pcmk__str_eq(src->exit_reason, dst->exit_reason, pcmk__str_none)) {
+        free(dst->exit_reason);
+        if (src->exit_reason == NULL) {
+            dst->exit_reason = NULL;
+        } else {
+            dst->exit_reason = strdup(src->exit_reason);
+        }
+    }
```

Comment:
seems as if we don't have a convenience function yet to get rid of that repeated logic.
pcmk__str_replace_cpy ...

Reply:
Yeah that bothered me too. Might as well create one ...

Reply:
added a couple commits for this